go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -359,5 +389,14 @@ func allocatePodAnnotationWithRollback(
 		podAnnotation = tentative
 	}
 
+	if ipamClaim != nil && claimsReconciler != nil && needsAnnotationUpdate && err == nil {
+		newIPAMClaim := ipamClaim.DeepCopy()
+		newIPAMClaim.Status.IPs = util.StringSlice(podAnnotation.IPs)
+		err = claimsReconciler.Reconcile(ipamClaim, newIPAMClaim, ipAllocator)
+		if err != nil {
+			releaseIPs = util.CopyIPNets(podAnnotation.IPs)
```

Comment:
For which case are we specifically doing this? This should only be done if IPs were actually allocated, and it was already done above.

Reply:
> For which case are we specifically doing this? This should only be done if IPs were actually allocated, and it was already done above.

Ah, I think I understand. When we allocate new IPs, we're already setting these as the `releaseIPs`. 

Right, this seems to be redundant.

Reply:
Done.